### PR TITLE
Use Crossdock V2

### PR DESCRIPTION
**Pull request by junchaowu**
@yarpc/java

### CODE REVIEW

**Comment by junchaowu** [comment]
I will fix coverall later.

**Comment by breerly** [comment]
I'm getting tired of coveralls breaking the build - can you disable or fix it?

File: yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java [hunk 1]
```diff
@@ -43,6 +43,9 @@
 public class HarnessServerInboundHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
     private static final Logger logger = LoggerFactory.getLogger(HarnessServerInboundHandler.class);
     private static final int remotePort = 8081;
+    private static final String SKIPED = "skiped";
```
Comment (kriskowal) [inline yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java]: skipped with two p's

File: yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java [hunk 2]
```diff
@@ -64,17 +67,32 @@ protected void messageReceived(ChannelHandlerContext channelHandlerContext, Full
 
         logger.debug("behavior " + behavior + " " + server);
 
-        String respBody = "not ok";
+        String respBody;
+        HarnessResponse harnessResponse = new HarnessResponse();
+        ;
```
Comment (kriskowal) [inline yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java]: semicolon stranded alone on a desert line. send help.

File: yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java [hunk 3]
```diff
@@ -64,17 +67,32 @@ protected void messageReceived(ChannelHandlerContext channelHandlerContext, Full
...
+        ObjectMapper objectMapper = new ObjectMapper();
         if (behavior.equals("echo")) {
-            EchoClient echoClient = new EchoClient(server, remotePort);
-            String hash = UUID.randomUUID().toString().substring(0, 7);
-            RPCFuture<Response<Echo>> future = echoClient.call(server, new Echo(hash));
-            Response<Echo> response = future.get();
-            if (response.getBody().getToken().equals(hash)) {
-                respBody = "ok - Server said: " + hash;
+            try {
```
Comment (kriskowal) [inline yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java]: consider breaking each behavior out into a function. there are going to be a few of these.

File: yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java [hunk 4]
```diff
@@ -64,17 +67,32 @@ protected void messageReceived(ChannelHandlerContext channelHandlerContext, Full
...
+                EchoClient echoClient = new EchoClient(server, remotePort);
+                String hash = UUID.randomUUID().toString().substring(0, 7);
+                Response<Echo> response = echoClient.call(server, new Echo(hash)).get();
+
+                if (response.getBody().getToken().equals(hash)) {
+                    harnessResponse.setStatus(PASSED);
+                } else {
+                    harnessResponse.setStatus(FAILED);
+                }
+                harnessResponse.setOutput(response.getBody().getToken());
```
Comment (kriskowal) [inline yarpc-core/src/test/java/io/yarpc/crossdock/HarnessServerInboundHandler.java]: `"Server said: " + `

**Comment by kriskowal** [comment]
changes suggested

**Pull request merged**